snippet_ranger/models/snippet.py

```diff
@@ -17,19 +17,30 @@ def construct(self, repository, filenames, sources, uasts,
         super(Snippet, self).construct(repository=repository, filenames=filenames,
                                        sources=sources, uasts=uasts)
         if ((positions_start is None) ^ (positions_end is None)) or \
-                ((positions_start is None) and (positions_end is None) ^ (positions is not None)):
+                ((positions_start is None) ^ (positions is not None)):
             raise ValueError("You should specify both positions_start and positions_end or "
                              "only a positions")
         if positions is None:
+            if len(positions_start) != len(positions_end):
+                raise ValueError("Length of positions_start ({}) and positions_end ({}) "
+                                 "are not equal".format(len(positions_start), len(positions_end)))
             self._positions = np.array(list(zip(positions_start, positions_end)))
         else:
             self._positions = positions
         return self
 
     @property
+    def names(self) -> list:
+        """
+        Creates list of names for each snippet.
```

Comment:
All right, then we need a more verbose explanation here, otherwise, folks like me will confuse the meaning